plugins/polls/app/controllers/polls_controller.rb

```diff
@@ -0,0 +1,93 @@
+class PollsController < ApplicationController
+  before_filter -> { require_plugin_enabled FoodsoftPolls }
+
+  def index
+    @polls = Poll.page(params[:page]).per(@per_page).order(created_at: :desc)
+  end
+
+  def show
+    @poll = Poll.find(params[:id])
+  end
+
+  def new
+    @poll = Poll.new
+  end
+
+  def create
+    @poll = Poll.new(params[:poll])
+    @poll.created_by = current_user
+
+    if @poll.save
+      redirect_to @poll, notice: t('.notice')
+    else
+      render action: 'edit'
+    end
+  end
+
+  def edit
+    @poll = Poll.find(params[:id])
+
+    if @poll.created_by != current_user && !current_user.role_admin?
+      redirect_to polls_path, alert: t('.no_right')
+    end
+  end
+
+  def update
+    @poll = Poll.find(params[:id])
+
+    if @poll.created_by != current_user && !current_user.role_admin?
```

Comment:
I'd favour moving this check to a (privat) method in the controller (also because it is repeated in `destroy`).